src/core/polyhedral/mapped_scop.cc

```diff
@@ -572,6 +573,9 @@ std::tuple<std::string, tc::Grid, tc::Block> MappedScop::codegen(
   code << "extern \"C\" {" << std::endl
        << emitCudaKernel(specializedName, *mappedScopForCodegen) << "}"
        << std::endl;
+#else
+  code << "TC was compiled with -DWITH_CUDA=0 which disabled cuda compilation";
```

Comment:
how about using `#error` to fail compilation with a meaningful message?

Reply:
This was a hotfix for CPU only build but it does not seem to affect anyone atm.
I have a much bigger refactoring in progress to clearly separate between: core, cpu, cuda and cross-compile cuda.
I'll revisit this PR in the bigger context and abandon for now.